core/trino-main/src/main/java/io/trino/sql/planner/EffectivePredicateExtractor.java

```diff
@@ -301,7 +301,12 @@ public Expression visitUnion(UnionNode node, Void context)
         @Override
         public Expression visitUnnest(UnnestNode node, Void context)
         {
-            return TRUE;
+            Expression sourcePredicate = node.getSource().accept(this, context);
+
+            return switch (node.getJoinType()) {
+                case INNER, LEFT -> pullExpressionThroughSymbols(sourcePredicate, node.getOutputSymbols());
```

Comment:
Where do you handle `LEFT` in printers?

Also I do not understand why we can pull predicate from sources for for INNER/LEFT but not the others.

Reply:
> Where do you handle LEFT in printers?

Through
```
label.append(node.getJoinType().getJoinLabel())
                        .append(" Unnest on true");
```
or similar logic.

Reply:
>why we can pull predicate from sources for for INNER/LEFT but not the others.

because RIGHT or FULL join can introduce nulls in replicate columns, and the nulls can potentially break predicates.